contracts/protocol/libraries/logic/PoolExtendedLogic.sol

```diff
@@ -0,0 +1,442 @@
+    using Math for uint256;
+    using UserConfiguration for DataTypes.UserConfigurationMap;
+    using PercentageMath for uint256;
+    using SafeCast for uint256;
+
+    event ReserveUsedAsCollateralDisabled(
+        address indexed reserve,
+        address indexed user
+    );
+
+    event BlurExchangeRequestInitiated(
+        address indexed initiator,
+        address paymentToken,
+        uint256 listingPrice,
+        uint256 borrowAmount,
+        address collection,
+        uint256 tokenId
+    );
+
+    event BlurExchangeRequestFulfilled(
+        address indexed initiator,
+        address paymentToken,
+        uint256 listingPrice,
+        uint256 borrowAmount,
+        address collection,
+        uint256 tokenId
+    );
+
+    event BlurExchangeRequestRejected(
+        address indexed initiator,
+        address paymentToken,
+        uint256 listingPrice,
+        uint256 borrowAmount,
+        address collection,
+        uint256 tokenId
+    );
+
+    function executeInitiateBlurExchangeRequest(
+        DataTypes.PoolStorage storage ps,
+        IPoolAddressesProvider poolAddressProvider,
+        DataTypes.BlurBuyWithCreditRequest[] calldata requests
+    ) external {
+        address keeper = ps._blurExchangeKeeper;
+        //check and update overall status
+        {
+            uint256 ongoingRequestAmount = ps._blurOngoingRequestAmount +
+                requests.length;
+            ValidationLogic.validateStatusForBlurExchangeRequest(
+                ps._blurExchangeEnable,
+                keeper,
+                ongoingRequestAmount,
+                ps._blurOngoingRequestLimit
+            );
+            ps._blurOngoingRequestAmount = ongoingRequestAmount.toUint8();
+        }
+
+        uint256 totalBorrow = 0;
+        address weth = poolAddressProvider.getWETH();
+        address oracle = poolAddressProvider.getPriceOracle();
+        DataTypes.UserConfigurationMap storage userConfig = ps._usersConfig[
+            msg.sender
+        ];
+
+        //validate request and mint nToken for every single request
+        {
+            uint256 remainingETH = msg.value;
+            uint256 requestFeeRate = ps._blurExchangeRequestFeeRate;
+            for (uint256 index = 0; index < requests.length; index++) {
+                DataTypes.BlurBuyWithCreditRequest calldata request = requests[
+                    index
+                ];
+                uint256 needCashETH = initiateBlurExchangeRequest(
+                    ps,
+                    userConfig,
+                    request,
+                    remainingETH,
+                    requestFeeRate,
+                    oracle
+                );
+                remainingETH -= needCashETH;
+                totalBorrow += request.borrowAmount;
+            }
+            require(remainingETH == 0, Errors.INVALID_ETH_VALUE);
+        }
+
+        //mint debt token
+        if (totalBorrow > 0) {
+            BorrowLogic.executeBorrow(
```

Comment:
should we do transferUnderlyingTo before executeBorrow ? since the borrow use rate   is not correct in updateInterestRates now. 
borrow use rate = total debt / (balance (pToken contract) + total debt)
balance (pToken contract) should subtract  the totalBorrow